src/test/java/htsjdk/samtools/CRAMComplianceTest.java

```diff
@@ -362,6 +371,81 @@ public void testBAMThroughCRAMRoundTripViaPath() throws IOException {
+        return new Object[][] {
+
+                // 1301_slice_aux.cram has legacy dataseries TC, TN present in it.
+                // TC, TN are now removed and are no longer written by htsjdk
+                // This file is used to test if CRAM reader can ignore these dataseries.
+                // 1301_slice_aux.cram is taken from hts-specs repo. It uses reference files, ce.fa and ce.fa.fai.
+                // 1301_slice_aux.cram file: https://github.com/samtools/hts-specs/blob/master/test/cram/3.0/passed/1301_slice_aux.cram
+                // Reference files: https://github.com/samtools/hts-specs/tree/master/test/cram
+                { new File(TEST_DATA_DIR, "1301_slice_aux.cram"),
+                        new File(TEST_DATA_DIR, "ce.fa") },
+        };
+    }
+
+    @Test(dataProvider = "obsoleteDataseriesCRAM", description = "round trip a CRAM file with obsolete dataseries TC and TN. TC and TN should not be present in the output CRAM file")
+    public void testRoundtripObsoleteDataseries(final File sourceFile, final File referenceFile) throws IOException {
+        final CRAMEncodingStrategy testStrategy = new CRAMEncodingStrategy();
+        final File tempOutCRAM = File.createTempFile("testRoundTrip", ".cram");
+        tempOutCRAM.deleteOnExit();
+        CRAMTestUtils.writeToCRAMWithEncodingStrategy(testStrategy, sourceFile, tempOutCRAM, referenceFile);
+        assertRoundTripFidelity(sourceFile, tempOutCRAM, referenceFile, false);
+        try (InputStream cramInputStream = new BufferedInputStream(new FileInputStream(tempOutCRAM));
+            CountingInputStream countingInputStream = new CountingInputStream(cramInputStream);
+            CramContainerIterator containerIterator = new CramContainerIterator(countingInputStream)) {
+            CramHeader cramHeader = containerIterator.getCramHeader(); //actual cram header
+            final long containerByteOffset = countingInputStream.getCount();
+            do {
+
+                // read the container and its CompressionHeader
+                Container container = new Container(cramHeader.getCRAMVersion(), countingInputStream, containerByteOffset);
+                CompressionHeader compressionHeader = container.getCompressionHeader();
+
+                // get the CompressionHeaderEncodingMap
+                CompressionHeaderEncodingMap encodingMap = compressionHeader.getEncodingMap();
+
+                //make sure obsolete DataSeries TC, TN are not present in the CompressionHeaderEncodingMap
+                for (final DataSeries dataseries : CompressionHeaderEncodingMap.DATASERIES_NOT_READ_BY_HTSJDK) {
+                    Assert.assertNull(encodingMap.getEncodingDescriptorForDataSeries(dataseries),
+                            "Unexpected encoding key found: " + dataseries.getCanonicalName());
+                }
+            } while(containerIterator.hasNext());
+        }
+    }
+
+    public void assertRoundTripFidelity(
```

Comment:
This can be removed (see my other comment about keeping the test focused on the specific issue of obsolete data series).